test/document_stores/test_document_store.py

```diff
@@ -119,6 +119,32 @@ def test_init_elastic_doc_store_with_index_recreation():
     assert len(labels) == 0
 
 
+@pytest.mark.elasticsearch
+def test_elasticsearch_eq_filter():
+    documents = [
+        {"content": "some text", "id": "1", "keyword_field": ["x", "y", "z"], "number_field": [1, 2, 3, 4]},
+        {"content": "some text", "id": "2", "keyword_field": ["x", "y", "w"], "number_field": [1, 2, 3]},
+        {"content": "some text", "id": "3", "keyword_field": ["x", "z"], "number_field": [2, 4]},
+        {"content": "some text", "id": "4", "keyword_field": ["x", "y"], "number_field": [2, 3]},
+    ]
+
+    index = "test_elasticsearch_eq_filter"
+    document_store = ElasticsearchDocumentStore(index=index, recreate_index=True)
+    document_store.write_documents(documents)
+
+    filter = {"keyword_field": {"$eq": ["z", "x"]}}
+    filtered_docs = document_store.get_all_documents(index=index, filters=filter)
+    assert len(filtered_docs) == 1
+    assert filtered_docs[0].meta["keyword_field"] == ["x", "z"]
+    assert filtered_docs[0].id == "3"
+
+    filter = {"number_field": {"$eq": [2, 3]}}
+    filtered_docs = document_store.query(query=None, index=index, filters=filter)
+    assert len(filtered_docs) == 1
+    assert filtered_docs[0].meta["number_field"] == [2, 3]
+    assert filtered_docs[0].id == "4"
```

Comment:
This part can be parametrized with something like:

```python
@pytest.mark.parametrize('field_name,field_value', [("keyword_field", ["z", "x"]), ("number_field", [2, 3]])
def test_elasticsearch_eq_filter(field_name, field_value):
    ....
```

Reply:
I tried but the two blocks are slightly different - they should probably go in two different tests but I think for this specific case and provided that this part will be heavily refactored soon and the test function is small enough, we can afford to keep it like this